### ISSUE TITLE
Decide on models (RNNs)

### ISSUE THREAD

**Issue by nathancoulson** [nathancoulson/prediction_app#2]
Research OOTB models to start with

LSTM stacked as base model.
1000 requests predicting the request mix of the next 250 requests as base prediction structure.

Metrics to tune for:

- Add validation loss
- Add app bias eval as part of grid search
- MAE as main loss

Issues to diagnose:

- [ ] Underfiting: looking for val loss gradient
- [ ] Overfitting: looking at training loss vs val lostt

Hyperparameters to tune:

- [ ] Input sequence length: (100, 500, 1000, 2000)
- [ ] Number of memory units: (10,25,50,75,100,200)
- [ ] Number of hidden layers: (1,2,5,8,10)
- [ ] Learning rate: (0.1, 0.01, 0.001, 0.0001)
- [ ] batch-size: (8, 16, 32, 64, 128)

Advanced (on final model):

regularisation:
dropout: (0.1, 0.2, 0.3, 0.4, 0.5)
L1L2  - if time permits

Early stopping:

Watch val-loss. Patience = 100

**Comment by nathancoulson** [comment]
Stacked LSTM and modified regression.
Done!
